Source/UI/NewUI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -3,120 +3,125 @@ import Checkout
-                case .city:
-                    updateSummaryValue(with: billingFormData?.address?.city, summaryValue: &summaryValue)
-                case .postcode:
-                    updateSummaryValue(with: billingFormData?.address?.zip, summaryValue: &summaryValue)
-                case .phoneNumber:
-                    updateSummaryValue(with: billingFormData?.phone?.number, summaryValue: &summaryValue, withNewLine: false)
-            }
-        }
-
-        paymentFormStyle?.editBillingSummary?.summary?.text = summaryValue
-        updateEditBillingSummaryView?()
+    let summary = updateSummaryValue(with: summaryValue)
+    guard !summary.isEmpty else {
+      let addBillingSummary = paymentFormStyle?.addBillingSummary ?? DefaultAddBillingDetailsViewStyle()
+      paymentFormStyle?.addBillingSummary = addBillingSummary
+      updateAddBillingDetailsView?()
+      return
     }
-
-    private func updateExpiryDate(){
-        updateExpiryDateView?()
+    paymentFormStyle?.editBillingSummary?.summary?.text = summary
+    updateEditBillingSummaryView?()
+  }
+
+  private func updateExpiryDate(){
+    updateExpiryDateView?()
+  }
+
+  private func isAddBillingSummaryNotUpdated() -> Bool{
+    guard billingFormData?.address != nil ||
+            billingFormData?.phone != nil else {
+      let addBillingSummary = paymentFormStyle?.addBillingSummary ?? DefaultAddBillingDetailsViewStyle()
+      paymentFormStyle?.addBillingSummary = addBillingSummary
+      updateAddBillingDetailsView?()
+      return false
     }
-
-    private func isAddBillingSummaryNotUpdated() -> Bool{
-        guard billingFormData?.address != nil ||
-              billingFormData?.phone != nil else {
-            let addBillingSummary = paymentFormStyle?.addBillingSummary ?? DefaultAddBillingDetailsViewStyle()
-            paymentFormStyle?.addBillingSummary = addBillingSummary
-            updateAddBillingDetailsView?()
-            return false
-        }
-        return true
-    }
-
-    private func updateSummaryValue(with value: String?, summaryValue: inout String,  withNewLine: Bool = true) {
-        guard let value = value else { return }
-        let billingFormValue = value.trimmingCharacters(in: .whitespaces)
-        if !billingFormValue.isEmpty {
-            let newLine = withNewLine ? "\n\n" : ""
-            summaryValue.append("\(billingFormValue)\(newLine)")
-        }
+    return true
+  }
+
+  private func updateSummaryValue(with summaryValues: [String?]) -> String {
+    let summaryValues =  summaryValues.compactMap { $0 }
+    var count = summaryValues.count
+    var result = ""
+    for value in summaryValues {
+      count -= 1
+      let billingFormValue = value.trimmingCharacters(in: .whitespaces)
+      if !billingFormValue.isEmpty {
+        let newLine = count !=  0 ? "\n\n" : ""
+        result.append("\(billingFormValue)\(newLine)")
+      }
     }
+    return result
```

Comment:
Is there a reason why we haven't done the following:

```suggestion
    let summaryValues =  summaryValues.compactMap { $0 }
    return summaryValues..joined(separator: "\n\n")
```